hr_switzerland/models/hr_payslip.py

```diff
@@ -37,10 +37,11 @@ def action_payslip_done(self):
                 move.action_post()
         return res
 
-    @api.onchange('employee_id', 'pay_13_salary', 'contract_id')
+    @api.depends('employee_id', 'pay_13_salary', 'contract_id')
     def _compute_13_salary(self):
         for payslip in self:
-            if payslip.pay_13_salary:
-                payslip.amount_13_salary = payslip.contract_id.provision_13_salary
-            else:
-                payslip.amount_13_salary = 0
+            if payslip.state == 'draft':
+                if payslip.pay_13_salary:
+                    payslip.amount_13_salary = payslip.contract_id.provision_13_salary
+                else:
+                    payslip.amount_13_salary = 0
```

Comment:
Compute methods should assign a value for every case otherwise it may crash.
```suggestion
            if payslip.state == 'draft':
                if payslip.pay_13_salary:
                    payslip.amount_13_salary = payslip.contract_id.provision_13_salary
                else:
                    payslip.amount_13_salary = 0
            else:
                payslip.amount_13_salary = ???  # To check
```

Reply:
@ecino 
if the payslip is not in draft, we don't want to change the amount_13_salary, if there was someting we don't want it to be changed

Reply:
could we assign the same value:
`payslip.amount_13_salary = payslip.amount_13_salary`

Reply:
No, because you change to compute method, all values will be erased, except if you add the `store=True` attribute

Reply:
I updated my suggestion.